apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Logging/GoogleLoggerTest.cs

```diff
@@ -55,9 +54,34 @@ private string Formatter(string message, Exception ex)
         [Fact]
         public void BeginScope()
         {
-            var logger = GetLogger(new Mock<IConsumer<LogEntry>>().Object);
-            // This will return null as the function is not implemented yet.
-            Assert.Null(logger.BeginScope("state"));
+            var expectedMessage = $"scope => {_logMessage}";
+            Predicate<IEnumerable<LogEntry>> matcher = (l) => l.Single().TextPayload == expectedMessage;
+            var mockConsumer = new Mock<IConsumer<LogEntry>>();
+            mockConsumer.Setup(c => c.Receive(Match.Create(matcher)));
+            var logger = GetLogger(mockConsumer.Object, LogLevel.Information);
+            using (logger.BeginScope("scope"))
+            {
+                logger.Log(LogLevel.Error, 0, _logMessage, null, Formatter);
+            }
+            mockConsumer.VerifyAll();
```

Comment:
What is this verifying? Sorry if this is a dumb question, but I haven't used the Moq stuff, so I've only partially gleaned how it's used from PR reviews.